reconcile/prometheus_rules_validator.py

```diff
@@ -0,0 +1,54 @@
+import sys
+import semver
+import logging
+
+import utils.gql as gql
+import utils.threaded as threaded
+import utils.promtool as promtool
+import reconcile.openshift_resources_base as orb
+import reconcile.queries as queries
+
+QONTRACT_INTEGRATION = 'prometheus_rules_validator'
+QONTRACT_INTEGRATION_VERSION = semver.format_version(0, 1, 0)
+
+
+def check_rule(rule):
+    try:
+        promtool.check_rule(rule['spec'])
+    except Exception as e:
+        return {'path': rule['path'], 'message': str(e).replace('\n', '')}
+
+
+def run(dry_run, thread_pool_size=10, cluster_name=None):
+    orb.QONTRACT_INTEGRATION = QONTRACT_INTEGRATION
+    orb.QONTRACT_INTEGRATION_VERSION = QONTRACT_INTEGRATION_VERSION
+
+    gqlapi = gql.get_api()
+    rules_paths = queries.get_prometheus_rules_paths()
+
+    rules = []
+    for n in gqlapi.query(orb.NAMESPACES_QUERY)['namespaces']:
+        if n['name'] != 'openshift-customer-monitoring':
+            continue
+
+        if cluster_name and n['cluster']['name'] != cluster_name:
+            continue
+
+        openshift_resources = n.get('openshiftResources')
+
+        for r in openshift_resources:
```

Comment:
I think `openshift_resources` can be `None`: https://gitlab.cee.redhat.com/service/app-interface/-/blob/67c684b180aaab5a5ef8b0dd038a9556a8f35b98/schemas/openshift/namespace-1.yml#L189

In that case, this may fail.

Reply:
fixed in a4b9b09